### PR TITLE
THORN-2031: Project rename changes.

### PR DESCRIPTION
**Pull request by mkouba**
- change groupId
- upgrade to config-api 1.4.0
- next version is 1.3.0
- update scm info

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -45,11 +46,10 @@
     <dependency>
       <groupId>junit</groupId>
       <artifactId>junit</artifactId>
-      <version>${version.junit}</version>
```
Comment (Ladicek) [inline pom.xml:48]: Looks suspicious?

Comment (Ladicek) [reply pom.xml:48]: Do we inherit that from parent?

Comment (mkouba) [reply pom.xml:48]: > The managed version is 4.12 The artifact is managed in org.wildfly.core:wildfly-core-parent:3.0.8.Final

And we did not specify `version.junit` in this pom...

Comment (Ladicek) [reply pom.xml:48]: Ah OK, thanks.

**Comment by mkouba** [comment]
Do not merge this yet. It seems the build passes but no classes are generated :(.

**Comment by scottmarlow** [comment]
I will verify locally, have to run a custom wf instance with the
wildfly-nosql subsystems defined, so we can extract the config-api...

**Comment by mkouba** [comment]
I see. Even the [release job](https://ci.wildfly-swarm.io/view/Config%20API/job/nosql-config-api-release) starts the instance before the build.

@scottmarlow Could you verify my PR pls?

**Comment by scottmarlow** [comment]
Changes looks good, verified that this is generating nosql config api Java files.

I could merge changes to wildfly-swarm/nosql-config-api (and possibly wildfly-swarm/wildfly-swarm, although I always created PRs).  Would be nice to have those privileges for thorntail also :)

**Comment by mkouba** [comment]
> Would be nice to have those privileges for thorntail also :)

@scottmarlow You'll have them ;-). We cannot add more members to thorntail org atm (Ken knows the details but it's connected to the fact that thorntail org was previously private).

**Pull request merged**